File: 0872-leaf-similar-trees/0872-leaf-similar-trees.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def leafSimilar(root1: TreeNode, root2: TreeNode): Boolean = {
        def getList(rt: TreeNode, ls: List[Int]):List[Int]={
            if (rt== null)
                ls
            else if(rt.left== null && rt.right== null)
                ls:+(rt.value)
            else{
                getList(rt.left, ls):::getList(rt.right, ls)    
            }
        }
        
        val ls1= getList(root1, List())
        val ls2= getList(root2, List())
        
        // ls1.map(v=> print(s"$v  "))
        // ls2.map(v=> print(s"$v  "))
        
        
        println(ls1)
        println(ls2)
        
        ls1== ls2
    }
}

File: 0872-leaf-similar-trees/0872-leaf-similar-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        
        
        List<Integer> ax1= getList(root1, new ArrayList<>());
        List<Integer> ax2= getList(root2, new ArrayList<>());
        
        
        System.out.println(ax1);
        System.out.println(ax2);
        return ax1.equals(ax2);
        
    }
    List<Integer> getList(TreeNode rt, List<Integer> ax){
        if(rt== null){
            return ax;
        }
        ax= getList(rt.left, ax);
        if(rt.left== null && rt.right== null){
            ax.addLast(rt.val);
        }
        ax= getList(rt.right, ax);
        
        return ax;
    }
}